### PR TITLE
nats: add KEMI publish function and event_callback param

### PR DESCRIPTION
**Pull request by seven1240**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description

- add `event_callback` KEMI param like in other modules
- expose `KSR.nats.publish`

I can fix module doc if it looks good. How to generate docs locally btw?

### CODE REVIEW

**Comment by henningw** [comment]
@seven1240 generate locally - see here http://www.kamailio.org/wiki/devel/module-docbook-readme

**Comment by seven1240** [comment]
I updated the doc and squashed into one commit.

**Comment by miconda** [comment]
@eschmidbauer: any comments on this PR?

File: src/modules/nats/nats_mod.c [hunk 1]
```diff
@@ -73,39 +79,36 @@ static void onMsg(
 		natsConnection *nc, natsSubscription *sub, natsMsg *msg, void *closure)
 {
 	nats_on_message_ptr on_message = (nats_on_message_ptr)closure;
-	char *s = (char *)natsMsg_GetSubject(msg);
+	// char *s = (char *)natsMsg_GetSubject(msg);
```
Comment (eschmidbauer) [inline src/modules/nats/nats_mod.c:82]: can we remove this comment?

File: src/modules/nats/nats_mod.c [hunk 2]
```diff
@@ -654,23 +664,33 @@ int _init_nats_sub_add(modparam_t type, void *val)
...
 	init_run_actions_ctx(&ctx);
-	run_top_route(event_rt.rlist[rt], fmsg, 0);
+	if (rt < 0 && keng) {
+		if (sr_kemi_route(keng, fmsg, EVENT_ROUTE,
+			&nats_event_callback, evname) < 0) {
+			LM_ERR("error running event route kemi callback\n");
+			}
+	} else {
+		run_top_route(event_rt.rlist[rt], fmsg, 0);
+	}
```
Comment (eschmidbauer) [inline src/modules/nats/nats_mod.c:693]:
```
if (rt < 0 && keng) {
    if (sr_kemi_route(keng, fmsg, EVENT_ROUTE, &nats_event_callback, evname) < 0) {
        LM_ERR("error running event route kemi callback\n");
    }
    return 0;
}

run_top_route(event_rt.rlist[rt], fmsg, 0);
return 0;
```

File: src/modules/nats/nats_mod.h [hunk 3]
```diff
@@ -30,6 +30,7 @@
 #include "../json/api.h"
 #include "../../core/cfg/cfg_struct.h"
 #include "../../core/fmsg.h"
+// #include "../../core/str.h"
```
Comment (eschmidbauer) [inline src/modules/nats/nats_mod.h:33]: can we remove this commented code ?

**Comment by seven1240** [comment]
Thanks. updated as suggested and squashed and rebased to the latest master.

**Comment by miconda** [comment]
I see @eschmidbauer approved the changes, therefore I merge it.

**Pull request merged**